### PR TITLE
Buffer packets for disconnected nodes

### PR DESCRIPTION
**Pull request by paulperegud**

### CODE REVIEW

File: golemproto/golem_protocol.py [hunk 1]
```diff
@@ -75,34 +100,23 @@ class acceptance(BaseProtocol.command):
...
+        cmd_id = 5
 
         structure = [
-            ('prefix', rlp.sedes.binary), # part added as a prefix to message, this is to prevent signing va banque check in blanco
-            ('signature', rlp.sedes.binary), # signature of of prefix concatenated with challenge
-            ('ethereum_pub_key', rlp.sedes.raw) # ethereum public key that node claims to have
-        ]
\ No newline at end of file
+            ('announcement_hash', rlp.sedes.big_endian_int),
+            ('hyperdrive_uri', rlp.sedes.binary), # Payload URI
```
Comment (Radagast-red) [inline golemproto/golem_protocol.py]: Hm this should be transport type agnostic whether it is Hyperdrive or IPFS

File: golemproto/golem_protocol.py [hunk 2]
```diff
@@ -42,7 +67,7 @@ class announcement(BaseProtocol.command):
         This is required because of need to be able to use single Ethereum account
         for multiple Golem nodes.
         """
-        cmd_id = 0
+        cmd_id = 2
 
         structure = [
             ('utc_time', rlp.sedes.binary), # string with ISO 8601 time representation
```
Comment (Radagast-red) [inline golemproto/golem_protocol.py]: What it is?

File: golemproto/golem_service.py [hunk 3]
```diff
@@ -243,17 +256,21 @@ def process_offers(self, ann):
...
-                log.warning("Can't ACK provider - can't reconnect back: {}, {}",
-                          offer.node_address, offer.node_id)
-                continue
-            log.warning("sending ack to {}".format(offer.node_id))
-            peer.protocols[GolemProtocol].send_acceptance(offer.offer_hash, 43,
-                                                          "signature")
+                packet = ('packet', 'create_acceptance', acc_short)
+                if not self.outbuff.has_key(peer.remote_pubkey):
+                    self.outbuff[peer.remote_pubkey] = deque([])
+                self.outbuff[peer.remote_pubkey].append(packet)
```
Comment (Radagast-red) [inline golemproto/golem_service.py]: so what if we never connect to peer? Does this will stay forever?

File: golemproto/golem_service.py [hunk 4]
```diff
@@ -100,36 +115,32 @@ def on_receive_announcement(self, proto, utc_time, price, requestor_id,
...
             proto.send_offer(*offer_short)
             cron.apply_after(OFFER_LOCK_TIMEOUT, self.mb_drop_obligation, offer)
         else:
-            self.ann = None
-            self.offer = None
-            self.ack = None
+            packet = ('packet', 'create_offer', offer_short)
+            if not self.outbuff.has_key(peer.remote_pubkey):
+                self.outbuff[peer.remote_pubkey] = deque([])
+            self.outbuff[peer.remote_pubkey].append(packet)
```
Comment (Radagast-red) [inline golemproto/golem_service.py]: when we remove this from outbuff?

File: golemproto/golem_protocol.py (hunk 2 above)
Comment (paulperegud) [inline golemproto/golem_protocol.py]: I've changed ids of all protocol messages to reorder them

File: golemproto/golem_service.py (hunk 4 above)
Comment (paulperegud) [inline golemproto/golem_service.py]: Yep, this is a memory leak. Timeouts is needed for outbuff values.

File: golemproto/golem_protocol.py (hunk 1 above)
Comment (paulperegud) [inline golemproto/golem_protocol.py]: We still need to know what should be called, Hyperdrive or IPFS. Explicit is better than implicit.

**Pull request merged**